webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueWriter.scala

```diff
@@ -79,12 +84,72 @@ class TechniqueWriter (
     archiver         : TechniqueArchiver
   , techLibUpdate    : UpdateTechniqueLibrary
   , translater       : InterpolatedValueCompiler
+  , readDirectives   : RoDirectiveRepository
+  , workflowLevelService: WorkflowLevelService
   , xmlPrettyPrinter : RudderPrettyPrinter
   , basePath         : String
 ) {
 
   private[this] var agentSpecific = new ClassicTechniqueWriter(basePath) :: new DSCTechniqueWriter(basePath, translater) :: Nil
 
+  def deleteTechnique(techniqueName : String, techniqueVersion : String, deleteDirective : Boolean, modId : ModificationId, committer : EventActor) : IOResult[Unit] ={
+
+    for {
+      d <- readDirectives.getFullDirectiveLibrary().map(_.allActiveTechniques.values.filter(_.techniqueName.value == techniqueName).flatMap(_.directives).filter(_.techniqueVersion.toString == techniqueVersion))
+      _ <- (d match {
+        case Nil => Right(d).toIO
+        case _ =>
+          if (deleteDirective) {
+
+            val wf = workflowLevelService.getWorkflowService()
+            for {
+              cr <- ((d.accumulate { directive =>
+                val tName = TechniqueName(techniqueName)
+                for {
+                  a <- readDirectives.getDirectiveWithContext(directive.id).notOptional(s"Could not find Directive ${directive.id.value}")
```

Comment:
I think that you should keep the `getFullDirectiveLibrary` and avoid to request here again. It will avoid ldap requests